src/mappings/subgraphService.ts

```diff
@@ -20,3 +22,148 @@ export function handleRewardsDestinationSet(event: RewardsDestinationSet): void
+    let indexerID = event.params.indexer.toHexString()
+    let allocationID = event.params.allocationId.toHexString()
+
+    // update indexer
+    let indexer = Indexer.load(indexerID)!
+    indexer.allocatedTokens = indexer.allocatedTokens.plus(event.params.tokens)
+    indexer.totalAllocationCount = indexer.totalAllocationCount.plus(BigInt.fromI32(1))
+    indexer.allocationCount = indexer.allocationCount + 1
+    indexer.save()
+
+    // update provision
+    let provision = createOrLoadProvision(event.params.indexer, event.address, event.block.timestamp)
+    provision.tokensAllocated = provision.tokensAllocated.plus(event.params.tokens)
+    provision.totalAllocationCount = provision.totalAllocationCount.plus(BigInt.fromI32(1))
+    provision.allocationCount = provision.allocationCount + 1
+    provision.save()
+
+    // update graph network
+    graphNetwork.totalTokensAllocated = graphNetwork.totalTokensAllocated.plus(event.params.tokens)
+    graphNetwork.allocationCount = graphNetwork.allocationCount + 1
+    graphNetwork.activeAllocationCount = graphNetwork.activeAllocationCount + 1
+    graphNetwork.save()
+
+    // update subgraph deployment
+    let deployment = createOrLoadSubgraphDeployment(subgraphDeploymentID, event.block.timestamp)
+    deployment.stakedTokens = deployment.stakedTokens.plus(event.params.tokens)
+    deployment.save()
+
+    // create allocation
+    let allocation = new Allocation(allocationID)
+    allocation.indexer = indexerID
+    allocation.provision = provision.id
+    allocation.creator = event.transaction.from
+    allocation.activeForIndexer = indexerID
+    allocation.activeForProvision = provision.id
+    allocation.subgraphDeployment = subgraphDeploymentID
+    allocation.allocatedTokens = event.params.tokens
+    allocation.effectiveAllocation = BigInt.fromI32(0)
+    allocation.createdAtEpoch = event.params.currentEpoch.toI32()
+    allocation.createdAtBlockHash = event.block.hash
+    allocation.createdAtBlockNumber = (
+        addresses.isL1 ? event.block.number : graphNetwork.currentL1BlockNumber!
+    ).toI32()
+    allocation.queryFeesCollected = BigInt.fromI32(0)
+    allocation.queryFeeRebates = BigInt.fromI32(0)
+    allocation.distributedRebates = BigInt.fromI32(0)
+    allocation.curatorRewards = BigInt.fromI32(0)
+    allocation.indexingRewards = BigInt.fromI32(0)
+    allocation.indexingIndexerRewards = BigInt.fromI32(0)
+    allocation.indexingDelegatorRewards = BigInt.fromI32(0)
+    allocation.delegationFees = BigInt.fromI32(0)
+    allocation.status = 'Active'
+    allocation.totalReturn = BigDecimal.fromString('0')
+    allocation.annualizedReturn = BigDecimal.fromString('0')
+    allocation.createdAt = event.block.timestamp.toI32()
+    allocation.indexingRewardCutAtStart = provision.indexingRewardsCut.toI32()
+    allocation.indexingRewardEffectiveCutAtStart = provision.indexingRewardsCut.toBigDecimal()
```

Comment:
this is incorrect no? we need to calculate with helper function

Reply:
Good catch, and yes, it's incorrect

Reply:
I think I added those helper functions in a separate PR though, will have to double check.

Another thing I was thinking is that updating everything PR by PR will be a rebase nightmare (as each PR depends on the previous, so a lot of rebases going to happen. I might simply aggregate all these changes, merge everything as is, and then add a bunch of PRs to fix each review separately 😓

Reply:
absolutely agree with putting everything into one big PR